content/Link Tracking with StimulusReflex - Part I.md

```diff
@@ -0,0 +1,152 @@
+# Link Tracking with StimulusReflex - Part I
+
+This is the introduction to a series on using StimulusReflex, a new tool to bring Rails to the
+era of the backend-managed-frontends. This might sound weird but there's been a couple of projects, starting with [Phoenix LiveView](https://hexdocs.pm/phoenix_live_view/Phoenix.LiveView.html) and following with things like [Motion](https://github.com/unabridged/motion) and [Sockpuppet](https://github.com/jonathan-s/django-sockpuppet), which use WebSockets to push updates from the server to the client and update the DOM accordingly.
+
+This is better defined by the [StimulusReflex's folks](https://docs.stimulusreflex.com) which as you may have guessed is a gem that does just that. I'm not going to explain this any further in words but rather I'll show you how to use it by building a link shortener with this
+new and exciting technology. In this part of the series, we'll focus on getting up and running so we can build awesome features for our project.
+
+![Header image](images/stimulus reflex - part I.jpeg)
+
+## Initializing the project
+
+To get started we are just going to follow the [original docs](https://docs.stimulusreflex.com/setup) and run the default `rails new` generator preconfigured to use [StimulusJS](https://stimulusjs.org) as the Javascript backend and the name of our project (`sho_lin` is short for **Sho**rtened **Lin**k, get it? I promise it's the first and only pun):
+
+```bash
+rails new sho_lin --webpack=stimulus -d postgresql
+```
+
+By the way, StimuluJS it's a frontend framework concerned with adding functionality to the HTML rather than controlling the whole DOM. If you wish to learn more about it, there's a [recent post](https://www.smashingmagazine.com/2020/07/introduction-stimulusjs/) that explains how it works.
```

Comment:
```suggestion
By the way, StimulusJS is a frontend framework concerned with adding functionality to the HTML rather than controlling the whole DOM. If you wish to learn more about it, there's a [recent post](https://www.smashingmagazine.com/2020/07/introduction-stimulusjs/) that explains how it works.
```